services/gateway/shared/registerHttpApi.js

```diff
@@ -64,137 +79,131 @@ const configureApi = (apiNames, apiPrefix, registeredModuleNames) => {
-				throw new ValidationException('Request param validation error.');
+				};
 			}
+		}
+		const apiPath = `${req.method.toUpperCase()} ${req.$alias.path}`;
+		return transformResponse(methodPaths[apiPath], data);
+	},
+});
 
-			const invalidList = paramReport.invalid;
-			if (invalidList.length > 0) {
-				sendResponse(INVALID_REQUEST[0], `Invalid input: ${invalidList.map(o => o.message).join('; ')}`);
-				throw new ValidationException('Request param validation error.');
-			}
-
-			const params = transformRequest(methodPaths[routeAlias], dropEmptyProps(paramReport.valid));
-			req.$params = params;
-		},
-
-		async onAfterCall(ctx, route, req, res, data) {
-			if (gatewayConfig.api.enableHttpCacheControl) {
-				// Set 'Cache-Control' to assist response caching
-				ctx.meta.$responseHeaders = { 'Cache-Control': gatewayConfig.api.httpCacheControlDirectives };
-			}
-
-			// Set response headers and return CSV data if filename available
-			if (data.data && data.meta && data.meta.filename && data.meta.filename.endsWith('.csv')) {
-				res.setHeader('Content-Disposition', `attachment; filename="${data.meta.filename}"`);
-				res.setHeader('Content-Type', 'text/csv');
-				res.end(data.data);
-				return res;
-			}
-
-			if (data.data && data.status) {
-				if (data.status === 'SERVICE_UNAVAILABLE') ctx.meta.$responseHeaders = { 'Retry-After': 30 };
-				if (data.status === 'INVALID_PARAMS') data.status = 'BAD_REQUEST';
+const registerApi = (apiNames, config, registeredModuleNames) => {
+	const allAPIs = getAllAPIs(apiNames, registeredModuleNames);
+	const apisToRegister = [];
 
-				ctx.meta.$statusCode = StatusCodes[data.status] || data.status;
+	const ALLOWED_ETAG_VALUES = new Set(['strong', 'weak', true, false]);
+	const DEFAULT_ETAG_VALUE = 'strong';
 
-				// 'ACCEPTED' is a successful HTTP code
-				if (data.status !== 'ACCEPTED') {
-					let message = `The request ended up with ${data.status} error`;
+	Object.entries(allAPIs).forEach(([key, apiConf]) => {
+		const eTag = apiConf.eTag === undefined ? DEFAULT_ETAG_VALUE : apiConf.eTag;
 
-					if (typeof data.data === 'object' && typeof data.data.error === 'string') {
-						message = data.data.error;
-					}
+		if (typeof eTag === 'function' || ALLOWED_ETAG_VALUES.has(eTag)) {
+			const apiConfig = buildAPIAliases(config.path, { key: allAPIs[key] });
+			apisToRegister.push(buildAPIConfig(`${config.path}${allAPIs[key].swaggerApiPath}`, config, apiConfig.aliases, apiConfig.whitelist, apiConfig.methodPaths, eTag));
+		} else {
+			logger.error(`etag not recognized for API: ${config.path}${allAPIs[key].swaggerApiPath}`);
+		}
+	});
 
-					return {
-						error: true,
-						message,
-					};
-				}
-			}
-			const apiPath = `${req.method.toUpperCase()} ${req.$alias.path}`;
-			return transformResponse(methodPaths[apiPath], data);
-		},
-	};
+	return apisToRegister;
```

Comment:
`registerApi` generates path for ready and status endpoints as `//ready` and `//status`. Please fix them.